tests/src/rules/no-duplicates.js

```diff
@@ -522,5 +522,99 @@ context('TypeScript', function () {
           }),
         ],
       });
+
+      if (!tsVersionSatisfies('>= 4.5') || !typescriptEslintParserSatisfies('>= 5.7.0')) {
+        return;
+      }
+
+      ruleTester.run('no-duplicates inline type', rule, {
```

Comment:
it'd be nice to use `[].concat()` and a ternary (adding the test cases or `[]`) rather than a totally separate `ruleTester.run`, but it's not a blocker for the PR.